src/app/public/modules/auth-http-client/auth-interceptor.ts

```diff
@@ -125,4 +126,37 @@ export class SkyAuthInterceptor implements HttpInterceptor {
     return next.handle(request);
   }
 
+  public getClient(handler: HttpHandler): HttpClient {
+    return new HttpClient(handler);
+  }
+
+  private getUrl(url: string, token: string, handler: HttpHandler): Promise<string> {
+    return this.config.runtime.command === 'serve' ?
+      this.getLocallyServedUrl(url, token, handler) :
+      this.getNonLocalUrl(url, token);
+  }
+
+  private getNonLocalUrl(requestUrl: string, token: string): Promise<string> {
+    const decodedToken = this.tokenProvider.decodeToken(token);
+    return BBAuthClientFactory.BBAuth.getUrl(requestUrl, {zone:  decodedToken['1bb.zone'] });
+  }
+
+  private getLocallyServedUrl(requestUrl: string, token: string, handler: HttpHandler): Promise<string> {
+    let match = TOKENIZED_URL_REGEX.exec(requestUrl);
+    if (match && match[LOCAL_PORT_INDEX]) {
+      let localPort = match[LOCAL_PORT_INDEX];
+      let client: HttpClient = this.getClient(handler);
+      return client.get(`http://localhost${localPort}/version`).toPromise()
```

Comment:
should the next iteration of this be to cache this somehow? if i'm reading this properly, this `/version` call would happen on _each_ call correct?

Reply:
we could, but this only affect local dev.  if we cache it then you don't get as responsive of a change to your local setup.

which our changes you could 
``` 
boot the UI --> talks to prod
boot your backend locally --> UI talks to backend
kill the backend --> UI is talking to prod again
```
if you push caching into it, then this is less responsive and slows down local dev which is what we're trying to avoid.

I've been strictly told the `serve` command is only utilized for local dev and thus allows us to isolate this non-caching check to local development only.